egs/wsj/s5/steps/nnet3/multilingual/combine_egs.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+#
+# This script generates examples for multilingual training of neural network
+# using separate input egs dir per language as input.
+# This scripts produces 3 sets of files --
+# egs.*.scp, egs.output.*.ark, egs.weight.*.ark
+#
+# egs.*.scp are the SCP files of the training examples.
+# egs.weight.*.ark map from the key of the example to the language-specific
+# weight of that example.
+# egs.output.*.ark map from the key of the example to the name of
+# the output-node in the neural net for that specific language, e.g.
+# 'output-2'.
+#
+# Begin configuration section.
+cmd=run.pl
+minibatch_size=512      # it is the number of consecutive egs that we take from 
+                        # each source, and it only affects the locality of disk 
+                        # access. This does not have to be the actual minibatch size;
+num_jobs=10             # helps for better randomness across languages
+                        # per archive.
+samples_per_iter=400000 # this is the target number of egs in each archive of egs
+                        # (prior to merging egs).  We probably should have called
+                        # it egs_per_iter. This is just a guideline; it will pick
+                        # a number that divides the number of samples in the
+                        # entire data.
+lang2weight=            # array of weights one per input languge to scale example's output
+                        # w.r.t its input language during training.
+stage=0
+
+echo "$0 $@"  # Print the command line for logging
+
+if [ -f path.sh ]; then . ./path.sh; fi
+. parse_options.sh || exit 1;
+
+num_langs=$1
+
+shift 1
+args=("$@")
+megs_dir=${args[-1]} # multilingual directory
+mkdir -p $megs_dir
+mkdir -p $megs_dir/info
+if [ ${#args[@]} != $[$num_langs+1] ]; then
+  echo "$0: num of input example dirs provided is not compatible with num_langs $num_langs."
+  echo "Usage:$0 [opts] <num-input-langs,N> <lang1-egs-dir> ...<langN-egs-dir> <multilingual-egs-dir>"
+  echo "Usage:$0 [opts] 2 exp/lang1/egs exp/lang2/egs exp/multi/egs"
+  exit 1;
+fi
+
+required="egs.scp combine.scp train_diagnostic.scp valid_diagnostic.scp"
+train_scp_list=
+train_diagnostic_scp_list=
+valid_diagnostic_scp_list=
+combine_scp_list=
+
+# read paramter from $egs_dir[0]/info and cmvn_opts
+# to write in multilingual egs_dir.
+check_params="info/feat_dim info/ivector_dim info/left_context info/right_context info/frames_per_eg info/final.ie.id cmvn_opts"
```

Comment:
I was commenting on this somewhere and it probably got lost. 
I think we should be careful in assuming info/final.ie.id exists -- I actually don't think we are there yet, so it would be better to check for the existence before copying it (or not cause the script to fail).